src/js/tabs/pid_tuning.js

```diff
@@ -355,6 +355,52 @@ TABS.pid_tuning.initialize = function (callback) {
             $('input[id="useIntegratedYaw"]').prop('checked', FC.ADVANCED_TUNING.useIntegratedYaw !== 0);
             //dmin column
             $('#pid_main .pid_titlebar2 th').attr('colspan', 6);
+
+            //dMinSwitch toggle
+            const dMinSwitch = $('#dMinSwitch');
+            dMinSwitch.prop('checked', FC.ADVANCED_TUNING.dMinRoll > 0 || FC.ADVANCED_TUNING.dMinPitch > 0 || FC.ADVANCED_TUNING.dMinYaw > 0);
+            dMinSwitch.change(function() {
+                const dMinChecked = $(this).is(':checked');
+                if (dMinChecked) {
+                    if (FC.TUNING_SLIDERS.slider_pids_mode != 0) {
+                        TuningSliders.sliderDMinRatio = 1;
+                    }
+                    if ($('.pid_tuning input[name="dMinRoll"]').val() == 0 && $('.pid_tuning input[name="dMinPitch"]').val() == 0 && $('.pid_tuning input[name="dMinYaw"]').val() == 0) {
+                        // when enabling dmin set its value based on 0.57x of actual dmax, dmin is limited to 100
+                        $('.pid_tuning input[name="dMinRoll"]').val(Math.min(Math.round($('.pid_tuning .ROLL input[name="d"]').val() * 0.57), 100));
+                        $('.pid_tuning input[name="dMinPitch"]').val(Math.min(Math.round($('.pid_tuning .PITCH input[name="d"]').val() * 0.57), 100));
+                        $('.pid_tuning input[name="dMinYaw"]').val(Math.min(Math.round($('.pid_tuning .YAW input[name="d"]').val() * 0.57), 100));
+                        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+                            $('.pid_tuning input[name="dMinRoll"]').val(Math.min(Math.round($('.pid_tuning .ROLL input[name="d"]').val() * 0.65), 100));
+                            $('.pid_tuning input[name="dMinPitch"]').val(Math.min(Math.round($('.pid_tuning .PITCH input[name="d"]').val() * 0.65), 100));
+                            $('.pid_tuning input[name="dMinYaw"]').val(Math.min(Math.round($('.pid_tuning .YAW input[name="d"]').val() * 0.65), 100));
+                        }
+                    } else {
+                        $('.pid_tuning input[name="dMinRoll"]').val(FC.ADVANCED_TUNING.dMinRoll);
+                        $('.pid_tuning input[name="dMinPitch"]').val(FC.ADVANCED_TUNING.dMinPitch);
+                        $('.pid_tuning input[name="dMinYaw"]').val(FC.ADVANCED_TUNING.dMinYaw);
+                    }
+                    $('.dMinDisabledNote').hide();
+                    $('.dminGroup .suboption').show();
+                    $('#pid_main tr :nth-child(5)').show();
+                    $('#pid_main .pid_titlebar2 th').attr('colspan', 6);
+                    $('.derivativeText').text(i18n.getMessage("pidTuningDMax"));
+                } else {
+                    if (FC.TUNING_SLIDERS.slider_pids_mode != 0) {
```

Comment:
This part is only available in firmware mode

Reply:
done